File: server/src/Connection/Messages/GameStatistics.cs
using System.Text.Json.Serialization;

namespace GameServer.Connection;

public record GameStatistics : Message
{
    [JsonPropertyName("messageType")]
    public override string MessageType { get; init; } = "GAME_STATISTICS";

    [JsonPropertyName("currentStage")]
    public string CurrentStage { get; init; } = "";

    [JsonPropertyName("countDown")]
    public int CountDown { get; init; } = 0;

    [JsonPropertyName("ticks")]
    public int Ticks { get; init; } = 0;

    [JsonPropertyName("scores")]
    public List<Score> Scores { get; init; } = new();

    public record scores
    {
        [JsonPropertyName("token")]
        public string Token { get; init; } = "";

        [JsonPropertyName("score")]
        public int Score { get; init; } = 0;
    }

}

File: sdk/python/src/agent/game_statistics.py
from typing import List
from .stage import Stage


class Score:
    def __init__(self, token: str, score: int):
        self.token = token
        self.score = score

    def __str__(self):
        return f"Score(token={self.token}, score={self.score})"


class GameStatistics:
    def __init__(
        self, current_stage: Stage, count_down: int, ticks: int, scores: List[Score]
    ):
        self.current_stage = current_stage
        self.count_down = count_down
        self.ticks = ticks
        self.scores = scores

    def __str__(self):
        scores_str = ", ".join(str(score) for score in self.scores)
        return f"GameStatistics(current_stage={self.current_stage}, count_down={self.count_down}, ticks={self.ticks}, scores=[{scores_str}])"
